### PR TITLE
Feature/i18n

### LINKED ISSUE #481: Add i18n support
In order to be able to present the user interface to people in their native languages, we need to internationalize our project.

Definition of done
---
* [x] Choose i18n framework.
* [x] Add basic i18n markup in UI templates for text strings.
* [ ] Document the logic for creating the string id's.

### PR DESCRIPTION
**Pull request by elnzv**
Closes #481

### CODE REVIEW

File: client/layouts/master_layout/sidebar.html [hunk 1]
```diff
@@ -3,52 +3,52 @@
   <aside class="main-sidebar">
     <section class="sidebar">
       <ul class="sidebar-menu">
-        <li class="header">Menu</li>
+        <li class="header">{{_ "sidebarMenu"}}</li>
```
Comment (brylie) [inline client/layouts/master_layout/sidebar.html]: Consider renaming these to include "masterLayout". E.g. "masterLayout-sidebarMenu", "masterLayout-sidebarAnalytics", ...

File: client/shared/not_authorized.html [hunk 2]
```diff
@@ -1,13 +1,13 @@
 <template name="notAuthorized">
   <div class="error-page">
-    <h1 class="headline text-yellow">Not Authorized</h1>
+    <h1 class="headline text-yellow">{{_ "notAuthorizedTitle"}}</h1>
```
Comment (brylie) [inline client/shared/not_authorized.html]: Separate the template name from the DOM element. E.g. "notAuthorized-Title", or "notAuthorized.title". Something to this effect.

File: client/shared/not_found.html [hunk 3]
```diff
@@ -2,10 +2,10 @@
   <div class="error-page">
     <h2 class="headline text-yellow"> 404</h2>
     <div class="error-content">
-      <h3><i class="fa fa-warning text-yellow"></i> Oops! Page not found.</h3>
+      <h3><i class="fa fa-warning text-yellow"></i> {{_ "notFoundTitle"}}</h3>
       <p>
-        We could not find the page you were looking for.<br>
-        Meanwhile, you may <a href="javascript:void(0)" onClick="history.go(-1); return false;">return to a previous page</a>.
+        {{_ "notFoundMessagePart1"}}<br>
```
Comment (brylie) [inline client/shared/not_found.html]: Separate each segment by dot or dash.. E.g. "notFound.message.part1" or "notFound-message-part1"

File: client/views/api_backends/api_backends.html [hunk 4]
```diff
@@ -1,39 +1,39 @@
...
+        <div>{{_ "apiBackendsHostRewrite"}}</div>
       </div>
       <div class="col-lg-4">
         {{> afQuickField name='backend_host'}}
       </div>
     </div>
-    <legend>Matching URL Prefixes</legend>
-    <span class="help-block">What URL prefixes should be routed to this backend?</span>
+    <legend>{{_ "apiBackendsMatchingTitle"}}</legend>
+    <span class="help-block">{{_ "apiBackendsMatchingTitleHelp"}}</span>
```
Comment (brylie) [inline client/views/api_backends/api_backends.html]: Sub elements still need separation. E.g. "apiBackends.matching.title.help" or "apiBackends-matching-title-help".

File: client/views/api_backends/api_backends.html [hunk 5]
```diff
@@ -48,18 +48,18 @@
       {{> afQuickField name='custom_rate_limits'}}
       {{/if}}
     </div>
-    <legend data-toggle="collapse" data-target="#advanced-rewriting-block" aria-expanded="false"><i class="fa fa-chevron-right"></i> Advanced Requests Rewriting</legend>
+    <legend data-toggle="collapse" data-target="#advanced-rewriting-block" aria-expanded="false"><i class="fa fa-chevron-right"></i> {{_ "apiBackendsAdvancedRequestsRewriting"}}</legend>
     <div id="advanced-rewriting-block" class="collapse in">
-      <span class="help-block">Modify the incoming request's URL or headers before passing it to the backend.</span>
+      <span class="help-block">{{_ "apiBackendsAdvancedRequestsRewritingHelp"}}</span>
```
Comment (brylie) [inline client/views/api_backends/api_backends.html]: Dashes or dots. Another e.g. "apiBackends.advanced.requestsRewriting.help" or "apiBackends-advanced-requestsRewriting-help". However they are specifically divided is up to you, but the divisions help us to understand and read the code.

File: client/shared/not_authorized.html [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
     <div class="error-content">
       <p>
         <i class="fa fa-warning text-yellow"></i>&nbsp;
-        You are not authorized to access this page. Please ensure you are logged in.
+        {{_ "notAuthorizedMessage"}}
```
Comment (brylie) [inline client/shared/not_authorized.html]: Separate "notAuthorized" and "message" by a dash (-). `"notAuthorized-message"`.

**Comment by brylie** [comment]
I added the WIP label, as there are still changes to be made with hyphenation of i18n IDs.

**Comment by brylie** [comment]
Also, be sure to resolve conflicts with `develop`.

**Comment by elnzv** [comment]
Ready to be merged.
Please, @apinf/developers, review one more time and merge it ASAP.

**Comment by bajiat** [comment]
@mauriciovieira or @frenchbread Can you please review the PR?

**Comment by frenchbread** [comment]
@bajiat I am currently reviewing #514, so if @mauriciovieira don't ming taking this, would be cool.

**Comment by mauriciovieira** [comment]
@bajiat, @frenchbread I will review this one.

On Tue, Oct 13, 2015 at 8:08 AM, Damir Mustafin <notifications@github.com>
wrote:

> @bajiat <https://github.com/bajiat> I am currently reviewing #514
> <https://github.com/apinf/api-umbrella-dashboard/pull/514>, so if
> @mauriciovieira <https://github.com/mauriciovieira> don't ming taking
> this, would be cool.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apinf/api-umbrella-dashboard/pull/515#issuecomment-147684584>
> .
>



-- 
Omne ignotum pro magnifico

Mauricio B. C. Vieira
http://mauriciovieira.net

**Comment by bajiat** [comment]
Thanks!

**Comment by mauriciovieira** [comment]
@elnzv Is there a strong reason for not using _, instead of -, like _masterLayout_sidebar_Menu_ ? I tested here with ':' and '.' separators but it does not work, but _ works. Anyways, it seems to be just a matter of style.

I think everything here is fine.

**Pull request merged**